springfox-core/src/main/java/springfox/documentation/schema/ModelProperty.java

```diff
@@ -115,4 +117,36 @@ public String getExample() {
   public String getPattern() {
     return pattern;
   }
+  
+  @Override
+  public int hashCode() {
+    return Objects.hashCode(name, type, qualifiedType, position, required, isHidden, 
+                            readOnly, description, allowableValues, getModelRef(), example, pattern);
+  }
+  
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+
+    ModelProperty that = (ModelProperty) o;
+    
+    return Objects.equal(name, that.name) &&
+        Objects.equal(type, that.type) &&
+        Objects.equal(qualifiedType, that.qualifiedType) &&
+        Objects.equal(position, that.position) &&
+        Objects.equal(required, that.required) &&
+        Objects.equal(isHidden, that.isHidden) &&
+        Objects.equal(readOnly, that.readOnly) &&
+        Objects.equal(description, that.description) &&
+        Objects.equal(allowableValues, that.allowableValues) &&
+        Objects.equal(getModelRef(), that.getModelRef()) &&
+        Objects.equal(example, that.example) &&
+        Objects.equal(pattern, that.pattern);
+  }
```

Comment:
Same as above. I'd rather extract this to an `Equivalence` [like this](https://github.com/springfox/springfox/blob/master/springfox-spring-web/src/main/java/springfox/documentation/spring/web/OperationCachingEquivalence.java).